packages/indexer-client/src/IndexerBaseClient.ts

```diff
@@ -109,6 +112,7 @@ export class IndexerBaseClient {
   readonly axiosInstance: AxiosInstance;
 
   constructor(opts: IndexerClientOpts) {
+    opts.v2Url = opts.v2Url ? opts.v2Url : opts.url.replace('v1', 'v2');
```

Comment:
Why would we need the override? Would it be safe to just derive this internally for now?

Not sure we should override opts, I feel like it should be immutable as much as possible.

Could we store this in a readonly variable similar to axiosInstance?

Reply:
check now